src/common/factory/emailDeliveryIntegrator/emailDeliveryIntegrator.factory.ts

```diff
@@ -0,0 +1,35 @@
+import { SendgridEmailDeliveryIntegrator } from './sendgridEmailDeliveryIntegrator';
+import { EmailType } from '../../../email/core/email-type';
+import {
+  EmailKeyHost,
+  EmailRegistration,
+} from '../../../email/core/register-email';
+import { SenderOptions } from 'src/email/core/sender-options';
+
+export interface IEmailDeliveryIntegrator {
+  sendEmail(
+    template: EmailType<EmailRegistration & EmailKeyHost>,
+    emailRegistration: EmailRegistration,
+    to: string,
+    locals: Record<string, string>,
+    options?: SenderOptions,
+    subject?: string,
+  ): Promise<void>;
+}
+
+export default class EmailDeliveryIntegratorFactory {
+  static getEmailDeliveryIntegrator(
+    emailDeliveryIntegratorName: string,
+  ): IEmailDeliveryIntegrator {
+    switch (emailDeliveryIntegratorName) {
+      case EmailDeliveryIntegratorName.SENDGRID:
+        return new SendgridEmailDeliveryIntegrator();
+      default:
+        return new SendgridEmailDeliveryIntegrator();
+    }
+  }
+}
+
+export enum EmailDeliveryIntegratorName {
```

Comment:
I tend to _avoid_ typescript enums. The reason is that two different enums with the _same value_ are not recognized as typescript as the same thing, even though the value is the same. This can give some composition problems.

I prefer POJOs (Plain Old Javascript Objects). Again, we can talk about this later if you want!